python/cuml/dask/common/base.py

```diff
@@ -18,27 +18,78 @@
 import numpy as np
 from toolz import first
 
-import cudf.comm.serialize  # noqa: F401
+from cuml.dask.common.utils import get_client
+from cuml.dask.common.input_utils import get_datatype
+import pickle
 
 from cuml import Base
 from cuml.common.array import CumlArray
 
 from dask_cudf.core import DataFrame as dcDataFrame
 
-from dask.distributed import default_client
 from functools import wraps
 
 
 class BaseEstimator(object):
 
-    def __init__(self, client=None, verbose=False, **kwargs):
+    def __init__(self, client=None, verbose=False, model=None, **kwargs):
         """
         Constructor for distributed estimators
         """
-        self.client = default_client() if client is None else client
+        self.client = get_client(client)
         self.verbose = verbose
         self.kwargs = kwargs
 
+        self.set_model(model)
+
+    @classmethod
+    def load(cls, file, client=None, verbose=False, pickle_args={}, **kwargs):
+        """
+        Convenience function to load a saved cuml model into a distributed
+        model.
+        """
+        model = pickle.load(file, **pickle_args)
+
+        if isinstance(model, Base):
+            # If serialized model is single GPU, create new
+            # dist model
+            model = cls.__init__(client=client,
+                                 verbose=verbose,
+                                 model=model,
+                                 **kwargs)
+        else:
+            model.client = get_client(client)
+            model.kwargs = kwargs
+            model.verbose = verbose
+        return model
+
+    def __getstate__(self):
+        inst_attrs = self.__dict__
+        inst_attrs["local_model"] = self.get_model()
+        del inst_attrs["client"]
+        return inst_attrs
+
+    def __setstate__(self, state):
+        self.__dict__.update(state)
+
+    def get_model(self):
+        """
+        Return trained single-GPU model
+        """
+        local_model = self.local_model
```

Comment:
I can rename it to something like `model_` maybe? Really the use of `local` wasn't meant to be local vs distributed but rather stating that it's the set of learned parameters encapsulated by the Dask model- thus it's state that is internal to the Dask model. Maybe we could call it `internal_model`?

Our base Dask wrapper kind of hides us from the details of `local_model` on purpose with the assumption that this model is generally assumed to be distributed unless there was a strict reason for bringing it local. This allows us to treat it in a unified manner, without having to manage multiple variables and perform complex state management. While the model's parameters are on device, it also concerns me that having multiple variables for this is going to lead to necessary (and unexpected) duplication of device memory. 

I'm hoping the `get_model()` function will provide unification in how we retrieve the `internal_model` so that users (and developers) who want to fetch the model directly (e.g., for serialization) can always expect the function to return it to the client. Currently, the `__getattr__` function on the base Dask model already exposes all the internal attributes of the `internal_model` by proxying local calls through it whether it's distributed or local to to the client. 

What do you think?